### ISSUE TITLE
The plugin stop tracking location randomly until user manually open the app again

### ISSUE THREAD

**Issue by ikleiman** [transistorsoft/capacitor-background-geolocation#156]
## Your Environment
* Plugin version: 4.10.0
* Platform: iOS and Android (provided logs are from Android device, but the client I'm developing this app for, has reported this same problem in iOS, before I've implemented the logger, if it happens again, will post those logs too)
* Device OS version: Android 13 (under MIUI 14)
* Device manufacturer / model: Xiaomi 11T Pro
* XCode version: -
* Capacitor info (`npx cap doctor`) 
````Latest Dependencies:

  @capacitor/cli: 4.7.1
  @capacitor/core: 4.7.1
  @capacitor/android: 4.7.1
  @capacitor/ios: 4.7.1

Installed Dependencies:

  @capacitor/core: 4.4.0
  @capacitor/cli: 4.4.0
  @capacitor/android: 4.4.0
  @capacitor/ios: 4.4.0
````
* Plugin config provided to `#ready`:
```javascript <!-- syntax-highlighting:  paste your code below -->
{
            // Geolocation Config
            desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_LOW,
            distanceFilter: 100,
            // Activity Recognition
            stopTimeout: 5,
            // Application config
            // debug: true, // <-- enable this hear sounds for background-geolocation life-cycle.
            logLevel: BackgroundGeolocation.LOG_LEVEL_DEBUG,
            logMaxDays: 7,
            stopOnTerminate: false,   // <-- Allow the background-service to continue tracking when user closes the app.
            startOnBoot: true,        // <-- Auto start tracking when device is powered-up.,
            url: environment.apiUrl + '/jwt/ubicacion/update',
            params: {
                tipo: this.platform.is('ios') ? 'ios' : 'android',
                userId: this.usuario.id
            },
            authorization: {
                strategy: 'JWT',
                accessToken: this.token
            },
            backgroundPermissionRationale: {
                title: 'Permitir a {applicationName} acceder a la ubicación del dispositivo en segundo plano?',
                message: 'Para poder mantener tu ubicación actualizada en segundo plano, por favor activa {backgroundPermissionOptionLabel} en los permisos de Ubicación. Esto es necesario para que la aplicación funcione correctamente y puedas compartir con tus amigos tu ubicación actual. Puedes desactivar esta funcionalidad en cualquier momento desde el menú de inicio.',
                positiveAction: 'Cambiar a {backgroundPermissionOptionLabel}',
                negativeAction: 'Cancelar'
            },
            locationAuthorizationAlert: {
                titleWhenOff: 'Activa tu ubicación en modo SIEMPRE',
                titleWhenNotEnabled: 'Activa tu ubicación en modo SIEMPRE',
                instructions: 'Para poder mantener tu ubicación actualizada en segundo plano, por favor activa "Siempre" en los permisos de Ubicación. Esto es necesario para que la aplicación funcione correctamente y puedas compartir con tus amigos tu ubicación actual. Puedes desactivar esta funcionalidad en cualquier momento desde el menú de inicio.',
                cancelButton: 'Cancelar',
                settingsButton: 'Configuración'
            },
            notification: {
                priority: BackgroundGeolocation.NOTIFICATION_PRIORITY_MIN,
                text: 'Actualizando tu ubicación',
                smallIcon: 'drawable/ic_stat_onesignal_default',
                largeIcon: 'drawable/ic_stat_onesignal_default',
            }
            // batchSync: true,
            // autoSyncThreshold: 5
        }
```

## Expected Behavior
Plugin should keep tracking location despite the user manually open the app or not

## Actual Behavior
If the user doesn't manually open the app, it randomly stops tracking location until it's manually opened again

## Steps to Reproduce
1. Init the plugin
2. Close the app and don't manually open it again
3. I don't know which one is the exact trigger, but the attached logs clearly shows the problem
4.

## Context
Trying to correctly track the device location without manually opening every few hours/days (and without rebooting the device).
I've manually opened the app on 03-22 16:10:36, took a flight from Santiago to Lima on 03-23 10:27:18, went out on destination (by car and by foot) after arrival on 03-23 13:46:12, took a flight back to Santiago on 03-24 16:20:03 - until this point everything worked perfectly. Landed on 03-24 19:46:44, turned off airplane mode, the app sent some location points from Lima airport and then stopped tracking at all. Took a 30km car ride back home and nothing, it didn't showed up the persistent notification either (it's important to mention that this same trip correctly shows up on Google Timeline). Two hour after that on 03-24 21:51:28 I've manually opened the app, got the correct location (but the app didn't sent it to the server) and the day after, on 03-25 12:49:12 went out by foot and the app started tracking back. So from 03-24 19:46:44 until 03-25 12:49:12, our server thought I was still at Lima airport, when in fact I was back home by 03-24 21:05.

The relevant time points in the logs have the same hours as described above.

UPDATE:
I don't know if it could be related, but I went out by foot again on 03-25 16:38:45 where the logs shows up something, but the plugin didn't tracked the first location point and sent it to the server until 03-25 16:45:03.

## Debug logs

<details>
	<summary>Logs</summary>

```<!-- Syntax highlighting:  DO NOT REMOVE -->
Github throws "There was an error creating your Issue: body is too long, body is too long (maximum is 65536 characters)."

Please download it from https://www.dropbox.com/s/rkatrnw6khy7bdh/background-geolocation%282%29.log.gz?dl=0
```

</details>

**Comment by christocracy** [comment]
- At `03-24 19:46`, the plugin correctly entered the stationary state and monitored a geofence around [that position](https://www.google.com/maps/place/33%C2%B023'22.6%22S+70%C2%B047'10.2%22W/@-33.3959468,-70.7991962,14.85z/data=!4m4!3m3!8m2!3d-33.389602!4d-70.786166) at the airport.  The device was only able to provide a poor accuracy location of `hAcc=1200.0`.

```
03-24 19:47:22.844 DEBUG [TSGeofenceManager startMonitoringStationaryRegion] 
  🎾  Start monitoring stationary region (radius: 150.0m -33.3896022,-70.7861659 hAcc=1200.0)
03-24 19:47:22.855 DEBUG [AbstractService start] 
  🎾  TrackingService [eventCount: 1]
03-24 19:47:22.855 INFO [TrackingService handleMotionChangeResult] 
╔═════════════════════════════════════════════
║ TrackingService motionchange: false
╠═════════════════════════════════════════════
```

- About 30s later, a `providerchange` change location was acquired (again with a poor accuracy of `hAcc=1299.999` meters.  The plugin reacquires the stationary geofence [here, at the other side of the airport](https://www.google.com/maps/place/33%C2%B023'13.1%22S+70%C2%B048'05.1%22W/@-33.3869677,-70.8036054,17z/data=!3m1!4b1!4m4!3m3!8m2!3d-33.3869722!4d-70.8014167).
```
╔═════════════════════════════════════════════
║ providerchange LocationResult: 5
╠═════════════════════════════════════════════
╟─ 📍  Location[fused -33.386979,-70.801404 hAcc=1299.999 et=+4d5h57m35s931ms vel=0.0], age: 1499ms, time: 1679698095614

03-24 19:48:17.119 INFO [TSLocationManager onSingleLocationResult] 
  🔵  Acquired providerchange position
03-24 19:48:17.128 DEBUG [LocationRequestService handleLocationResult] SingleLocationRequest 5 isFinished? true
03-24 19:48:17.128 DEBUG [AbstractService finish] ⚙️︎  finish LocationRequestService [eventCount: 0, sticky: false]
03-24 19:48:17.130 INFO [b persist] 
  ✅  INSERT: 2354e899-3dd4-4306-b10a-79aba4cc3b75
03-24 19:48:17.130 DEBUG [AbstractService finish] ⚙️︎  finish LocationRequestService [eventCount: 0, sticky: false]
03-24 19:48:17.131 DEBUG [TSLocationManager$b onLocation] 
  ℹ️  Distance from last location: 2452658.2
03-24 19:48:17.132 DEBUG [TSGeofenceManager startMonitoringStationaryRegion] 
  🎾  Start monitoring stationary region (radius: 150.0m -33.3869792,-70.8014035 hAcc=1299.999)
```

The plugin will automatically initiate tracking when the device exits that `150m` geofence.  Geofences are monitored by the OS -- It's up to the OS to fire a geofence event if the device exits that area or the motion-activity API (also monitored by the OS) reports that the device is in-motion (`on_foot`, `in_vehicle`, `on_bicycle`, etc).

Sometimes, due to the environment and/or the device, things just don't work.

**Comment by ikleiman** [comment]
Thanks, what are the possible reasons for Google Timeline to be correctly tracking the airport-home trip, while the plugin didn't? Just the geofence? Is there any setting that could be tested against this problem? 

I can keep providing the logs when this happen again with other devices, but for instance I remember a few weeks ago something similar happened with my own device: took a car from point A to point B (around 1 linear KM), the plugin didn't tracked the trip, 2 hours later, went by foot from point B to point A and less than a minute from the start point B, the persistent notification showed up and the plugin started tracking the trip correctly (without opening the app or doing anything else).

I understand (and totally agree) from the technical point of view that sometimes thing just don't work, but if it's not possible to keep this situations at minimal occurrences with another settings, I'll need to explain to my client with simple words why.

**Comment by christocracy** [comment]
> Google Timeline to be correctly tracking the airport-home trip, while the plugin didn't? Just the geofence?

Geofence + motion api did not trigger.

> Is there any setting that could be test against this problem

no

I do not have any issues like this on my devices.  See https://dontkillmyapp.com

Your problem is likely related to your particular device.  There are many Android devices that have problems (as seen at https://dontkillmyapp.com)

**Comment by christocracy** [comment]
> I guess that if an app gets killed it doesn't autostart later by its own?

Yes it does.

**Comment by ikleiman** [comment]
Today it happened again, with an iPhone 8 (iOS 14.8) this time. My client was at point A, we met at point B and I was at point C.

My device correctly tracked from point C to B and then back from B to C. His device correctly tracked from point A (2023-03-28 11:11:35) to point B (2023-03-28 11:48:23), then got stuck at point B. When he was back to point A, he manually opened the app (2023-03-28 13:36:24) and some location points from 40 minutes before that time-point (when he started the trip back) started being sent out to the server. It's important to mention that from point A to point B he was constantly using the app every few minutes, then for the trip back I've asked him to keep it closed the entire trip to check the background behavior.

Please find the logs on this link: https://www.dropbox.com/s/od2jiu533fb1xi1/background-geolocation%283%29.log.gz?dl=0

**Comment by christocracy** [comment]
Your DB data is not interesting to me.  I suggest you fetch the plug-in logs and see what the plug-in was saying in its logs during any times of interest.

**Comment by christocracy** [comment]
See wiki “Debugging” to learn about using the .emailLog method to fetch plug-in logs.

**Comment by ikleiman** [comment]
That was the second message as support for the first one that has the plugin logs, fetched with .emailLog as seen on "Debugging" on the Wiki.

First message: https://github.com/transistorsoft/capacitor-background-geolocation/issues/156#issuecomment-1487309709
Second message: https://github.com/transistorsoft/capacitor-background-geolocation/issues/156#issuecomment-1487336448

**Comment by christocracy** [comment]
> `logLevel: BackgroundGeolocation.LOG_LEVEL_DEBUG`


When you have a problem, use `LOG_LEVEL_VERBOSE`.

**Comment by christocracy** [comment]
When you see the following in the logs, your app has been re-launched after being terminated (terminated by the user, the OS, or your app crashed)

```
2023-03-28 13:38:03.862 ℹ️-[TSLocationManager init] 
╔═════════════════════════════════════════════
║ TSLocationManager (build 388)
╠══════════════════════════════════════════════
{

```

**Comment by ikleiman** [comment]
Thanks, I've just uploaded the new version with VERBOSE level. 

Meanwhile, why if the app is being terminated, is still recording the location points, but not uploading them unless the user manually opens the app? 

And then it kept uploading them (I guess that in the same queue order) for more than an hour: from 2023-03-28 13:36:25 to 2023-03-28 14:43:32 it kept uploading the corresponding points from 2023-03-28 12:56:00 to 2023-03-28 13:29:00.

Also it has a 100m distanceFilter, and on trip from B to A (not working as expected) recorded around 10-15 different location points per minute on a car trip, while on trip from A to B (worked as expected), it recorded 1-3 max different location points per minute. Same conditions for both trips. Does that tells anything about the problem?

**Comment by christocracy** [comment]
I suggest you post a map view of your points here or configure the plug-in to upload locations to the demo server at tracker.transistorsoft.com (see api docs TransistorAuthorizationToken).

**Comment by ikleiman** [comment]
Sure,

Trip from point A to point B, 29 minutes, 54 location points uploaded real time:

![Captura de pantalla 2023-03-29 084049](https://user-images.githubusercontent.com/4931407/228527147-b9b9392e-5086-4b05-8e1f-343697c9ae98.png)


Trip from point B to point A, 37 minutes (includes a 5 minute walk to his car), 434 location points uploaded over 4.5 hours (the upload started when the user manually opened the app, when he got back at point A):

![Captura de pantalla 2023-03-29 084504](https://user-images.githubusercontent.com/4931407/228527189-e8fcef12-a42b-4d75-a646-44c311bddacb.png)

**Comment by ikleiman** [comment]
Here's another case of the same problem, iPhone XR iOS 16.3.1 this time. Please take a look to 2023-04-11 12:37, until that point everything worked as expected that day. Then the user didn't opened the app until the next log entry 2023-04-12 19:09. At that point and until 2023-04-12 20:22, it started to upload hundreds of location points corresponding from 2023-04-11 13:10:00 to 2023-04-12 19:02:00. Then it suddenly stopped uploading, until the next time the user manually opened the app at 2023-04-13 18:16:05 when started to upload the rest of that trip from 2023-04-12 19:02:00 to 2023-04-12 19:20:00.

Here are some screens of the trips the user made, and their upload time:

![imagen](https://user-images.githubusercontent.com/4931407/232637774-e48b09ac-36a7-48b5-92fb-6619ae4b1ea8.png)

![imagen](https://user-images.githubusercontent.com/4931407/232638195-5abcad20-849b-434b-911c-a2a83f7ee994.png)

![imagen](https://user-images.githubusercontent.com/4931407/232638413-a0be593a-1db6-4704-9160-4cb51ad0eb0d.png)

![imagen](https://user-images.githubusercontent.com/4931407/232638610-36ce9b0d-5299-47ac-8dfa-c7ab112b6175.png)

![imagen](https://user-images.githubusercontent.com/4931407/232639059-8d07e0bb-7e42-4319-ac7a-4b58f7966336.png)

![imagen](https://user-images.githubusercontent.com/4931407/232639561-6a20ed06-6c64-479f-975a-a477b7dea6bb.png)

Please find the log on this link: https://www.dropbox.com/s/mt6cy018utsyuoh/background-geolocation%2810%29.log.gz?dl=0

When the upload finished, it worked OK for some points and then began the exact same behavior, starting at 2023-04-14 20:03:40, as seen on this second log file: https://www.dropbox.com/s/mpvwxwa9oqqufs2/background-geolocation%2811%29.log.gz?dl=0

Please let me know if you need any further information.

**Comment by ikleiman** [comment]
@christocracy did you had a chance to check on this issue? It seems to be working fine in Android, but it's randomly showing up this behavior across Apple devices. I have two more cases on two different iPhone's, in case you need those logs. 

It's always the same: the route it's being correctly recorded, but it's not uploaded to the server unless the user manually open the app. Also, when this behavior appears, it begins to record way more location points than it should (currently it's configured for 150m and in car trips the location points are being recorded about every 80m or even less).

**Comment by christocracy** [comment]
I suggest you install the /example app in this repo onto your device and compare its performance.

**Comment by christocracy** [comment]
> (currently it's configured for 150m and in car trips the location points are being recorded about every 80m or even less).

Your logs show the plugin is clearly configured for `distanceFilter = 75`

If your app enters the background while the plugin is in the *stationary state* (that is, where location-services are OFF), iOS provides *exactly* 30s of background-time for you app to remain alive.

If the plugin has a large amount of records queued up in its SQLite database while in that state, the OS is only going to allow the plugin to upload for 30s.  The next time the app comes to foreground, the plugin's HTTP service will resume.

Also, the plugin is subscribed to background-fetch events which will also periodically allow the plugin's HTTP service to resume for 30s blocks of time.

One odd thing about your logs I see is a constant loss of `speed` reading between locations (`speed 3.86 mps`, `speed -1.00 mps`).  When `speed` shows `-1` and `accuracy 40-60m, it means that the location likely didn't come from GPS (likely came from Wifi).

```
2023-04-10 15:49:31.174 
📍<-33.40064475,-70.54767574> +/- 6.32m (speed 3.86 mps / course 209.97) @ 10-04-23 15:49:31 Chile Standard Time
.
.
.
2023-04-10 15:49:44.692 
📍<-33.40197558,-70.54836166> +/- 40.00m (speed -1.00 mps / course -1.00) @ 10-04-23 15:49:44 Chile Standard Time
.
.
.
```

- The plugin auto-scales `distanceFilter` based upon speed.  
- When there is no `speed` available (speed is provided by GPS satellites), the plugin must reset its `distanceFilter` to the configured value (`75` in your case).

I don't think this is a problem with the plugin -- I think it's related to the environment.

**Comment by ikleiman** [comment]
Thanks Chris, that's very useful information. If you take a look to the last log, you can see that the distanceFilter have been changed from 75 to 150. In the screens set (with 75m configured), when the problem shows up, the distance goes down to about 30m. But in the last log, with 150m configured as distance filter, it goes down to about 80m.

We're testing different configurations in order to study different behaviors, right now we're going to test it with distanceFilter=250, disableElasticity=true and DESIRED_ACCURACY_HIGH. I'm waiting for the AppStore release of that version.

During the week I've been doing some DB analysis also. One key finding is that the problem started right away after the release of the first version on the AppStore. For instance, for a given person, using the same device, same user, same settings, same environment, same daily routes, etc, it gave less than 1% of delays on the uploaded location points while we were testing the app over TestFlight during a 7 weeks period. After the release to the AppStore, for that same person/device, it gave 29% of delays on the uploaded points for a similar 7 weeks period. 

Do you have any idea what could be causing something like that on the release version?

About half of the Apple users shows that same behavior: over TestFlight, 10 users and all of them with less than 1% (there was just one user with 2.5% as maximum). Over AppStore, 18 users, 17% of delays in average, with 1% minimum and 48% maximum. 10 users with less than 5% and the other 8 with 28% in average.

**Comment by christocracy** [comment]
> Do you have any idea what could be causing something like that on the release version?

I have no idea.  This plug-in has thousands of customers (with presumably millions of device installs).  Nobody else is reporting this sort of thing.

iOS 16.4 made some changes.  If I were you, I would configure `showsBackgroundLocationIndicator: true`.

If I were concerned with too much data, i would definitely not configure `disableElasticity: true`

**Comment by ikleiman** [comment]
Thanks, I've added showsBackgroundLocationIndicator: true to the next version.

I have the results of the last version test. With this last information we can assure that there's definitively something wrong with the plugin, at least while this behavior is present. I understand that the other 1000's customers aren't reporting this problem, but that doesn't make it disappear by it's own. 

I've added an additional API POST (using Angular Http) inside the onLocation event and that one is working flawless until now at least (less than 1% delays). 

As you can see on the following screens, this is the same trip, uploaded by the plugin's built-in API POST mechanism: 

![Captura de pantalla 2023-05-01 170025](https://user-images.githubusercontent.com/4931407/235715907-58046872-5c8d-42d4-8395-a8524a7ac2eb.png)

And then the one uploaded by the API call inside onLocation event: 

![Captura de pantalla 2023-05-01 165957](https://user-images.githubusercontent.com/4931407/235715937-eedc2f5d-7762-42ab-abef-d911aa81f9a1.png)

The built-in API POST didn't started uploading the trip until the user manually opened the app at 2023-04-30 16:46:08, while the onLocation event API POST started right away when the user started the trip at 2023-04-30 15:35:51. 

Also there was nothing being logged between 2023-04-30 11:53:44 and 2023-04-30 16:46:07, despite I'm making a direct call to "debug" logger method inside onLocation event (I've added some custom messages in order to track onLocation and onHttp events). 

Beside that, the points are being recorded about each 60m, while the configuration is 250m with disableElasticity=true. But when the plugin works correctly, it does record about 250m (all the screens are from the same user, same device, same trip, even same day):

![Captura de pantalla 2023-05-01 190448](https://user-images.githubusercontent.com/4931407/235715993-e1ed7ca9-0fc7-4546-a0a8-e5950e2bafd6.png)

So in short, when this behavior appears, neither the plugin built-in API POST nor the plugin logger nor the distanceFilter/disableElasticity settings are working, while Angular Http POST is working without problems inside onLocation event.

The same has already happened in 3 different devices.

Please find the respective logs here: https://www.dropbox.com/s/nefe6wstb8d8nlp/background-geolocation%2813%29.log.gz?dl=0

Any ideas? Could you add up something on your side in order to debug this issue? And thanks in advance, as a developer myself, I know that this kind of issues are the worst ones.

**Comment by christocracy** [comment]
I suggest you install the /example app in this repo and compare its results.

the /example app is configured to post data to my demo server at tracker.transistorsoft.com.

**Comment by ikleiman** [comment]
How that would help at all to solve this issue? There's something failing inside the plugin here, that's seems to be caused by my particular development, so please lets focus on that.

I have another theory that would explain this behavior, there's any scenario where the plugin takes the default configuration parameters instead of the ones provided inside the ready method? That would explain the built-in API POST, the logger and the distanceFilter/disableElasticity not working as expected (no api, logger off and default distance/elasticity).

**Comment by christocracy** [comment]
> How that would help at all to solve this issue? There's something failing inside the plugin here, 

I don’t believe that’s the case until I see it reproduced in isolation of your app.  So please go ahead and install the /example app.

**Comment by tanuj-g** [comment]
I am also facing the exact same scenarios whatever mentioned above with some IOS devices (IPhone 14)

@ikleiman did you find the solution of it?

**Comment by christocracy** [comment]
See wiki "Debugging".  Learn to fetch and analyze the logs.  Also search api docs "Logger".

**Comment by tanuj-g** [comment]
@christocracy I already read the debugging in wiki. 
let me go through the logs generated by plugin and will get back to you. 

One thing which i didn't find in the docs is that, if user gave all the location permissions in the ios expect the motion activity 
then what happend, how we find in the plugin event that user didn't gave the motion activity permission. 

like in provider change event we have this object: 

`"provider": {
            "network": true,
            "enabled": false,
            "accuracyAuthorization": 1,
            "gps": true,
            "status": 3
        }`
        
        Which key indicates that user didn't gave the motion activity permission and denied it

**Comment by christocracy** [comment]
Motion permission is not required for the plugin to operate.  Motion permission has nothing to with the "Location Provider" thus does not appear in the `onProviderChange` event.

**Comment by tanuj-g** [comment]
[background-geolocation.log (1).zip](https://github.com/transistorsoft/capacitor-background-geolocation/files/13367879/background-geolocation.log.1.zip)


@christocracy 
here is the debugging log of my app,

At line 3567, there is a log entry for the HTTP Service with a count of 38. This seems to indicate that 38 location entries have been stored in the SQL database. However, you're puzzled as to why these entries were saved without corresponding HTTP API calls.

Line 3578 provides more insight with a response code of 0, accompanied by an error message stating that the host "api.server.com" could not be resolved because no address is associated with the hostname. Despite this error, you’ve noted that your API appears to be functioning correctly.

Lastly, line 3889 mentions that the HTTP service is busy, followed by an HTTP 200 response, which typically indicates success. Subsequently, only one record is deleted, yet the log suggests that all 38 records were transmitted sequentially.

Can you please help me to understand these debugging logs so I can find out why my tracking stopped in b/w and then started in the middle but old locations data came first and then the latest

**Comment by ikleiman** [comment]
@tanuj-g yes, not completely, but managed to reduce the delays to less than 1%. Not quite sure what specifically did the trick, but first I've restructured my app in order to be more like the /example app, in the following way:

1. Moved my BG plugin logic from app.component.ts to home.page.ts (my app logic is login => home => (other logged user pages), so home is the first logged user reachable page)
2. My API works with a JWT token that I was saving on a "this.token" variable inside home.page.ts, changed that to a function that always retrieves the token directly from the device storage

Then, in addition to the Http post feature of the plugin, I've implemented my own upload function inside onLocation event, then on the server I'm ordering the location points according the timestamp and skipping the duplicates using that information too. You must be aware that this approach is not the most efficient in battery matters, but was the only one that solved my delay scenarios.

**Comment by christocracy** [comment]
HTTP Requests are *synchronous*, the next one executes only after the previous returns a response of `20x`.  Look further in your logs to see the remaining HTTP requests occuring.

```
11-11 11:22:18.494 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:18.495 DEBUG [b destroy] 
  ✅  DESTROY: 4b49f7c3-6226-4d24-9250-1553e4ebec19
11-11 11:22:18.501 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:18.503 INFO [HttpService a] 
  🔵  HTTP POST: b0e2c8ab-300d-4542-8ae2-11bf144f52ec
11-11 11:22:18.555 DEBUG [AbstractService onDestroy] 
  🔴  TrackingService stopped
11-11 11:22:18.890 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:18.891 DEBUG [b destroy] 
  ✅  DESTROY: b0e2c8ab-300d-4542-8ae2-11bf144f52ec
11-11 11:22:18.899 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:18.901 INFO [HttpService a] 
  🔵  HTTP POST: ceb43c85-f15a-4f7e-9c9a-a446a54602d9
11-11 11:22:19.332 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:19.338 DEBUG [b destroy] 
  ✅  DESTROY: ceb43c85-f15a-4f7e-9c9a-a446a54602d9
11-11 11:22:19.348 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:19.349 INFO [HttpService a] 
  🔵  HTTP POST: 8ea166ed-d0cf-493d-b74d-55528675cdca
11-11 11:22:19.772 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:19.775 DEBUG [b destroy] 
  ✅  DESTROY: 8ea166ed-d0cf-493d-b74d-55528675cdca
11-11 11:22:19.782 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:19.785 INFO [HttpService a] 
  🔵  HTTP POST: 47740109-7afb-4055-84d5-1d486f73e440
11-11 11:22:20.153 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:20.156 DEBUG [b destroy] 
  ✅  DESTROY: 47740109-7afb-4055-84d5-1d486f73e440
11-11 11:22:20.164 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:20.166 INFO [HttpService a] 
  🔵  HTTP POST: 60dad6ad-b8dd-4bf6-bde9-104a81f1a350
11-11 11:22:20.541 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:20.543 DEBUG [b destroy] 
  ✅  DESTROY: 60dad6ad-b8dd-4bf6-bde9-104a81f1a350
11-11 11:22:20.552 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:20.554 INFO [HttpService a] 
  🔵  HTTP POST: 843dc1fe-231c-4ffd-ab6a-adbbb5d3016c
11-11 11:22:20.974 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:20.976 DEBUG [b destroy] 
  ✅  DESTROY: 843dc1fe-231c-4ffd-ab6a-adbbb5d3016c
11-11 11:22:20.982 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:20.983 INFO [HttpService a] 
  🔵  HTTP POST: 49b95065-09dd-4db1-b8c3-5f517ba950c0
11-11 11:22:21.335 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:21.337 DEBUG [b destroy] 
  ✅  DESTROY: 49b95065-09dd-4db1-b8c3-5f517ba950c0
11-11 11:22:21.351 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:21.353 INFO [HttpService a] 
  🔵  HTTP POST: 3eecfcdf-5fed-43d9-bdf2-8437576857c5
11-11 11:22:21.707 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:21.711 DEBUG [b destroy] 
  ✅  DESTROY: 3eecfcdf-5fed-43d9-bdf2-8437576857c5
11-11 11:22:21.720 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:21.721 INFO [HttpService a] 
  🔵  HTTP POST: 31553f26-172e-442b-8571-5ad775c46a69
11-11 11:22:22.278 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:22.280 DEBUG [b destroy] 
  ✅  DESTROY: 31553f26-172e-442b-8571-5ad775c46a69
11-11 11:22:22.295 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:22.298 INFO [HttpService a] 
  🔵  HTTP POST: 0a20520c-9b7e-47fd-9341-c662a1440c44
11-11 11:22:22.665 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:22.669 DEBUG [b destroy] 
  ✅  DESTROY: 0a20520c-9b7e-47fd-9341-c662a1440c44
11-11 11:22:22.679 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:22.682 INFO [HttpService a] 
  🔵  HTTP POST: 0144cf70-2855-4498-8794-86872916309e
11-11 11:22:23.142 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:23.144 DEBUG [b destroy] 
  ✅  DESTROY: 0144cf70-2855-4498-8794-86872916309e
11-11 11:22:23.148 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:23.149 INFO [HttpService a] 
  🔵  HTTP POST: 889031d2-e6b5-47c7-b133-2fb546eae3e3
11-11 11:22:23.604 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:23.606 DEBUG [b destroy] 
  ✅  DESTROY: 889031d2-e6b5-47c7-b133-2fb546eae3e3
11-11 11:22:23.615 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:23.616 INFO [HttpService a] 
  🔵  HTTP POST: 811ae9da-fb52-4e44-bff4-bc8a215ee514
11-11 11:22:23.956 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:23.960 DEBUG [b destroy] 
  ✅  DESTROY: 811ae9da-fb52-4e44-bff4-bc8a215ee514
11-11 11:22:23.968 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:23.970 INFO [HttpService a] 
  🔵  HTTP POST: e3badb9c-37c5-40e2-9ba4-df654cafe48c
11-11 11:22:24.441 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:24.442 DEBUG [b destroy] 
  ✅  DESTROY: e3badb9c-37c5-40e2-9ba4-df654cafe48c
11-11 11:22:24.451 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:24.454 INFO [HttpService a] 
  🔵  HTTP POST: 0c93b805-5219-4b08-bfa2-6a77a3c2107c
11-11 11:22:24.892 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:24.894 DEBUG [b destroy] 
  ✅  DESTROY: 0c93b805-5219-4b08-bfa2-6a77a3c2107c
11-11 11:22:24.902 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:24.904 INFO [HttpService a] 
  🔵  HTTP POST: 8abe0155-b260-4ec2-b15a-be3d51d762d4
11-11 11:22:25.267 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:25.271 DEBUG [b destroy] 
  ✅  DESTROY: 8abe0155-b260-4ec2-b15a-be3d51d762d4
11-11 11:22:25.280 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:25.281 INFO [HttpService a] 
  🔵  HTTP POST: 72bcbeb8-3605-4b18-a2f1-60d7328b0924
11-11 11:22:25.597 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:25.600 DEBUG [b destroy] 
  ✅  DESTROY: 72bcbeb8-3605-4b18-a2f1-60d7328b0924
11-11 11:22:25.608 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:25.610 INFO [HttpService a] 
  🔵  HTTP POST: f6e57958-1ad9-49f6-aefb-7ee9670a28db
11-11 11:22:25.991 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:25.993 DEBUG [b destroy] 
  ✅  DESTROY: f6e57958-1ad9-49f6-aefb-7ee9670a28db
11-11 11:22:26.003 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:26.005 INFO [HttpService a] 
  🔵  HTTP POST: 407c4b1a-9491-48ba-9ec7-0615a7153121
11-11 11:22:26.500 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:26.502 DEBUG [b destroy] 
  ✅  DESTROY: 407c4b1a-9491-48ba-9ec7-0615a7153121
11-11 11:22:26.514 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:26.517 INFO [HttpService a] 
  🔵  HTTP POST: feb6ce09-97a2-497d-b83d-b142d126a6fe
11-11 11:22:27.014 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:27.017 DEBUG [b destroy] 
  ✅  DESTROY: feb6ce09-97a2-497d-b83d-b142d126a6fe
11-11 11:22:27.024 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:27.027 INFO [HttpService a] 
  🔵  HTTP POST: 27e60287-4af2-40a0-aca7-6faeaec0ab0d
11-11 11:22:27.530 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:27.535 DEBUG [b destroy] 
  ✅  DESTROY: 27e60287-4af2-40a0-aca7-6faeaec0ab0d
11-11 11:22:27.542 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:27.543 INFO [HttpService a] 
  🔵  HTTP POST: 463abd22-fe12-45a6-a109-1cdf543fa7cd
11-11 11:22:28.138 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:28.139 DEBUG [b destroy] 
  ✅  DESTROY: 463abd22-fe12-45a6-a109-1cdf543fa7cd
11-11 11:22:28.147 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:28.148 INFO [HttpService a] 
  🔵  HTTP POST: 112b3b32-a295-450d-91f7-5f577fe6a669
11-11 11:22:28.640 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:28.643 DEBUG [b destroy] 
  ✅  DESTROY: 112b3b32-a295-450d-91f7-5f577fe6a669
11-11 11:22:28.650 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:28.653 INFO [HttpService a] 
  🔵  HTTP POST: 3e359b0f-9353-423f-94c6-44cc7929dbd9
11-11 11:22:29.069 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:29.074 DEBUG [b destroy] 
  ✅  DESTROY: 3e359b0f-9353-423f-94c6-44cc7929dbd9
11-11 11:22:29.079 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:29.080 INFO [HttpService a] 
  🔵  HTTP POST: 76d936b2-fa95-4d79-a3fe-6f777fe69f85
11-11 11:22:29.457 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:29.459 DEBUG [b destroy] 
  ✅  DESTROY: 76d936b2-fa95-4d79-a3fe-6f777fe69f85
11-11 11:22:29.475 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:29.479 INFO [HttpService a] 
  🔵  HTTP POST: c1763ca0-3f32-45a1-859d-9e37c4654a2b
11-11 11:22:29.880 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:29.883 DEBUG [b destroy] 
  ✅  DESTROY: c1763ca0-3f32-45a1-859d-9e37c4654a2b
11-11 11:22:29.891 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:29.893 INFO [HttpService a] 
  🔵  HTTP POST: aa75d24b-3de6-4cfa-8530-09ee4c8e706f
11-11 11:22:30.288 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:30.290 DEBUG [b destroy] 
  ✅  DESTROY: aa75d24b-3de6-4cfa-8530-09ee4c8e706f
11-11 11:22:30.295 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:30.298 INFO [HttpService a] 
  🔵  HTTP POST: 056fa162-627d-4385-bf6b-b79b937ef2b5
11-11 11:22:30.810 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:30.814 DEBUG [b destroy] 
  ✅  DESTROY: 056fa162-627d-4385-bf6b-b79b937ef2b5
11-11 11:22:30.823 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:30.825 INFO [HttpService a] 
  🔵  HTTP POST: 3da5a53a-f9bb-4278-9bfc-91f44ce9f984
11-11 11:22:31.265 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:31.270 DEBUG [b destroy] 
  ✅  DESTROY: 3da5a53a-f9bb-4278-9bfc-91f44ce9f984
11-11 11:22:31.284 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:31.286 INFO [HttpService a] 
  🔵  HTTP POST: 574193a2-0274-4b23-8534-b84dbefe443d
11-11 11:22:31.730 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:31.734 DEBUG [b destroy] 
  ✅  DESTROY: 574193a2-0274-4b23-8534-b84dbefe443d
11-11 11:22:31.743 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:31.746 INFO [HttpService a] 
  🔵  HTTP POST: d0e12300-b857-4430-a4a3-3de60da8c9e2
11-11 11:22:32.161 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:32.166 DEBUG [b destroy] 
  ✅  DESTROY: d0e12300-b857-4430-a4a3-3de60da8c9e2
11-11 11:22:32.173 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:32.174 INFO [HttpService a] 
  🔵  HTTP POST: 59d3dc76-b650-4752-94c6-200e7455e994
11-11 11:22:32.750 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:32.751 DEBUG [b destroy] 
  ✅  DESTROY: 59d3dc76-b650-4752-94c6-200e7455e994
11-11 11:22:32.755 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:32.756 INFO [HttpService a] 
  🔵  HTTP POST: c13da0c6-27d2-4bdd-912c-e1947b23b081
11-11 11:22:33.102 INFO [HttpService$h onResponse] 
  🔵  Response: 200
11-11 11:22:33.104 DEBUG [b destroy] 
  ✅  DESTROY: c13da0c6-27d2-4bdd-912c-e1947b23b081
11-11 11:22:33.112 DEBUG [b first] 
  ✅  Locked 1 records
11-11 11:22:33.118 INFO [HttpService a] 
  🔵  HTTP POST: 4c5b1f80-bf23-4fe5-bc0d-8b3544eda700
11-11 11:22:33.389 DEBUG [c h] 
  ℹ️  LocationAuthorization: Permission granted
11-11 11:22:33.420 DEBUG [AbstractService start] 
  🎾  LocationRequestService [eventCount: 1]
11-11 11:22:33.422 INFO [SingleLocationRequest startUpdatingLocation] 
  🔵  [SingleLocationRequest start, action: 2, requestId: 6]
11-11 11:22:33.425 DEBUG [AbstractService finish] ⚙️︎  finish LocationRequestService [eventCount: 0, sticky: true]
11-11 11:22:33.650 INFO [HttpService$h onResponse] 
  🔵  Response: 200
.
.
.
```

**Comment by tanuj-g** [comment]
@christocracy,  first and foremost, thank you for your time and assistance.

I encountered a situation where the plugin abruptly ceased transmitting location updates in the background until I opened the app manually. Referring to the log, at line 5123 with the timestamp 11-17 16:27:59.704, there's a message indicating that the BackgroundTaskService stopped. Subsequently, there are no further log entries, nor any background updates. Despite the user travelling around 5 miles, no updates were received during this period.

Later, at 11-17 17:14:36.000, when I manually activated the app, an update was immediately recorded, as evident in the subsequent log entry at line 5125.

I'm puzzled about what occurred post-16:27, especially since there are no corresponding log entries to shed light on the issue.

`╔═════════════════════════════════════════════
║ TSLocationManager version: 3.4.0 (420)
╠═════════════════════════════════════════════
╟─ realme RMX2156 @ 12 (capacitor)
{
  "activityRecognitionInterval": 10000,
  "allowIdenticalLocations": false,
  "authorization": {},
  "autoSync": true,
  "autoSyncThreshold": 0,
  "backgroundPermissionRationale": {
    "title": "Allow Chaine to access to this device's location in the background?",
    "message": "Chaine needs location permission when you are assigned load to provide updates. This app collects location data to enable sharing tracking updates with your customers even when the app is closed or not in use.",
    "positiveAction": "Change to {backgroundPermissionOptionLabel}",
    "negativeAction": "Cancel"
  },
  "batchSync": false,
  "configUrl": "",
  "debug": false,
  "deferTime": 0,
  "desiredAccuracy": -2,
  "desiredOdometerAccuracy": 100,
  "disableAutoSyncOnCellular": false,
  "disableElasticity": false,
  "disableLocationAuthorizationAlert": true,
  "disableMotionActivityUpdates": false,
  "disableProviderChangeRecord": false,
  "disableStopDetection": false,
  "distanceFilter": 200,
  "elasticityMultiplier": 4,
  "enableHeadless": false,
  "enableTimestampMeta": false,
  "extras": {
    "androidSDKVersion": 31,
    "appVersion": "3.0.10",
    "batteryLevel": 0.699999988079071,
    "connectionType": "wifi",
    "deviceID": "5b484dd8eabde232",
    "deviceManufacturer": "realme",
    "deviceModel": "RMX2156",
    "deviceName": "realme narzo 30",
    "deviceOS": "android",
    "deviceOSVersion": "12",
    "devicePlatform": "android",
    "driverPhone": "4052955384",
    "iOSVersion": null,
    "isDeviceCharging": false,
    "isDeviceVirtual": false,
    "isNetworkConnected": true,
    "languageCode": "en",
    "userID": "eb0713b8-010e-49ed-981e-c1698999b909"
  },
  "fastestLocationUpdateInterval": -1,
  "foregroundService": true,
  "geofenceInitialTriggerEntry": true,
  "geofenceModeHighAccuracy": false,
  "geofenceProximityRadius": 1000,
  "geofenceTemplate": "",
  "headers": {
    "ContentType": "application\/json"
  },
  "headlessJobService": "com.chaineapp.driver.BackgroundGeolocationHeadlessTask",
  "heartbeatInterval": -1,
  "httpRootProperty": "location",
  "httpTimeout": 60000,
  "isMoving": true,
  "locationAuthorizationRequest": "Always",
  "locationTemplate": "",
  "locationTimeout": 60,
  "locationUpdateInterval": 1000,
  "locationsOrderDirection": "ASC",
  "logLevel": 5,
  "logMaxDays": 3,
  "maxBatchSize": -1,
  "maxDaysToPersist": 1,
  "maxRecordsToPersist": -1,
  "method": "POST",
  "minimumActivityRecognitionConfidence": 75,
  "motionTriggerDelay": 0,
  "notification": {
    "layout": "",
    "title": "",
    "text": "Location Service activated",
    "color": "",
    "channelName": "TSLocationManager",
    "channelId": "",
    "smallIcon": "",
    "largeIcon": "",
    "priority": 0,
    "sticky": false,
    "strings": {},
    "actions": []
  },
  "params": {},
  "persist": true,
  "persistMode": 2,
  "schedule": [],
  "scheduleUseAlarmManager": false,
  "speedJumpFilter": 300,
  "startOnBoot": true,
  "stationaryRadius": 25,
  "stopAfterElapsedMinutes": 0,
  "stopOnStationary": false,
  "stopOnTerminate": false,
  "stopTimeout": 5,
  "triggerActivities": "in_vehicle, on_bicycle, on_foot, running, walking",
  "url": "https:\/\/api.chaineapp.com\/apps\/events\/v1\/mobile\/chaine\/locations",
  "useSignificantChangesOnly": true,
  "enabled": true,
  "schedulerEnabled": false,
  "trackingMode": 1,
  "odometer": 0,
  "isFirstBoot": false,
  "didLaunchInBackground": false,
  "didDeviceReboot": false
}`

The settings are being output in the log. Could you help me understand what might have happened, and explain the log's content that I might be missing or not fully comprehending at the moment?

[log.gz](https://github.com/transistorsoft/capacitor-background-geolocation/files/13434592/log.gz)

**Comment by christocracy** [comment]
When you see this, your app process has been relaunched after being terminated.

```
11-17 17:14:36.001 INFO [LoggerFacade$a a] 
╔═════════════════════════════════════════════
║ TSLocationManager version: 3.4.0 (420)
╠═════════════════════════════════════════════
╟─ realme RMX2156 @ 12 (capacitor)
{
```

When the device is detected to be moving, the OS will automatically relaunch your app process to resume tracking.

If your device is sitting stationary for long periods of time, the OS can and will terminate your app process to free up memory for the app currently running in the foreground.  If the device moves about 200 meters from last known location, tracking **WILL** resume, even if the OS terminated your app.

**Comment by tanuj-g** [comment]
> When you see this, your app process has been relaunched after being terminated.
> 
> ```
> 11-17 17:14:36.001 INFO [LoggerFacade$a a] 
> ╔═════════════════════════════════════════════
> ║ TSLocationManager version: 3.4.0 (420)
> ╠═════════════════════════════════════════════
> ╟─ realme RMX2156 @ 12 (capacitor)
> {
> ```
> 
> When the device is detected to be moving, the OS will automatically relaunch your app process to resume tracking.
> 
> If your device is sitting stationary for long periods of time, the OS can and will terminate your app process to free up memory for the app currently running in the foreground. If the device moves about 200 meters from last known location, tracking **WILL** resume, even if the OS terminated your app.

@christocracy 
The above should be expected behaviour but my app is not relaunched by itself I am moving for almost 6-7 miles still app is not restarted and plugin didn't send any location update. 

at 11-17 17:14:36.001 - I opened the app manually to restart the tracking 
is there any scenario where app will not restart itself if the user is moving

**Comment by christocracy** [comment]
@tanuj-g post a new issue.
